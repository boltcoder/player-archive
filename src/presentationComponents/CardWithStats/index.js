import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { Avatar, Grid, CircularProgress } from '@material-ui/core';
import stylesheet from './style';
import { Typography, Stats } from 'appPresentationComponents';

const useStyles = makeStyles(stylesheet);

// Quintessential example of a presentation component written completely decoupled from business logic
// No notion of "Player" info leaks in this. 
// this very well to show a certain Dish Info with calorie stats
const CardWithStats = (props) => {
  const classes = useStyles();
  const {
    title,
    subtitle,
    stats,
    loading,
  } = props;

  const StatsComponent = stats.map(({
    label,
    value,
  }, index)=>(
    <Grid key={`${title}_card_${index}`} item xs={12} md={6} >
      <Stats 
        label={label}
        value={value}
      />
    </Grid>
  ));
  return (
    <div className={classes.container}>
      <Grid container direction="column" alignItems="center">
        <Avatar>{title[0]}</Avatar>
        <Typography color="primaryText" variant="h6" weight="medium">{title}</Typography>
        <Typography color="primaryText" variant="subtitle2">{subtitle}</Typography>
        <Grid container>

          {StatsComponent}
        </Grid>
      </Grid>
      {loading && <div className={classes.loading}><CircularProgress color="primary" size={40}></CircularProgress></div>}
    </div>
  );
}

CardWithStats.propTypes = {
  title: PropTypes.oneOf([PropTypes.string,PropTypes.number,PropTypes.element]),
  subtitle: PropTypes.oneOf([PropTypes.string,PropTypes.number,PropTypes.element]),
  stats: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.oneOf([PropTypes.string,PropTypes.number,PropTypes.element]),
    value: PropTypes.oneOf([PropTypes.string,PropTypes.number,PropTypes.element]),
  })),
  loading: PropTypes.bool,
}


CardWithStats.displayName = 'CardWithStats';
export default CardWithStats;