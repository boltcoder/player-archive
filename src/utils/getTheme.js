import { createMuiTheme, responsiveFontSizes} from '@material-ui/core/styles';

const getTheme = ({ themeType }) => {

  const theme = createMuiTheme({
    palette: {
      type: themeType,
    },  
    flexboxes: {
      horizontallyVerticallyCentered: {
        display: 'flex',
        alignItems:'center',
        justifyContent: 'center',
      },
      verticallyCentered: {
        display: 'flex',
        alignItems:'center',
      },
      horizontallyCentered: {
        display: 'flex',
        justifyContent:'center',
      }
    }
  });
  return responsiveFontSizes(theme,{factor:3});
}
export default getTheme;