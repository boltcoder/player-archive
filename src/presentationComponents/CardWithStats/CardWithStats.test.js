import React from 'react';
import CardWithStats from './index.jsx';
import { MuiThemeProvider } from '@material-ui/core/styles';
import getTheme from 'appUtils/getTheme';
import renderer from 'react-test-renderer';
import { render, fireEvent} from '@testing-library/react';

const ThemeWrappedCardWithStats = (props) => <MuiThemeProvider theme={getTheme({themeType: "light"})}><CardWithStats {...props} /></MuiThemeProvider>;


describe("Testing CardWithStats", ()=> {

  it("Variant should render correct tags",()=>{
    const htmlTree = renderer.create(<ThemeWrappedCardWithStats title="test1" subtitle="test2" stats={[{label:"Age",value:12}] } />).toJSON;
    expect(htmlTree).toMatchSnapshot();


  })

  

})