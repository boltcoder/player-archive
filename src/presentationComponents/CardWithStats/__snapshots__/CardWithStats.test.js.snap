// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CardWithStats Variant should render correct tags 1`] = `[Function]`;
